src/main/java/eu/ai4work/sws/service/RuleEngineService.java

```diff
@@ -49,54 +43,76 @@ public SlidingDecisionResult applySlidingDecisionRules(Map<String, Object> slidi
-        // (which will be greater than -1.0) will replace this initial value.
-        double highestMembershipDegree = -1.0;
-
-        for (Map.Entry<String, LinguisticTerm> termEntry : suggestedWorkSharingApproachAsFuzzyVariable.getLinguisticTerms().entrySet()) {
-            double membershipDegree = termEntry.getValue().getMembershipFunction().membership(suggestedWorkSharingApproachAsFuzzyVariable.getLatestDefuzzifiedValue());
-            if (membershipDegree > highestMembershipDegree) {
-                highestMembershipDegree = membershipDegree;
-                linguisticTerm = termEntry.getKey();
-            }
-        }
+    private String mapFuzzyInferenceResultToLinguisticTerm(Variable suggestedWorkSharingApproachAsFuzzyVariable) throws UnIdentifiedTermException {
+        return suggestedWorkSharingApproachAsFuzzyVariable.getLinguisticTerms().entrySet().stream()
+                // Map each linguistic term to its corresponding membership degree
+                // The key is the linguistic term name
+                // The value is membership degree for the latest defuzzified value
+                .map(linguisticTermWithMembershipDegree -> Map.entry(
+                        linguisticTermWithMembershipDegree.getKey(),
+                        linguisticTermWithMembershipDegree.getValue().getMembershipFunction()
+                                .membership(suggestedWorkSharingApproachAsFuzzyVariable.getLatestDefuzzifiedValue())
+                ))
+                // Identify the linguistic term with the highest membership degree
+                .max(Map.Entry.comparingByValue())
+                // Retrieve the linguistic term with the highest membership degree, if available
+                .map(Map.Entry::getKey)
+                // If no valid term is found, throw an exception
+                .orElseThrow(() -> new UnIdentifiedTermException(
+                        "Unable to identify the linguistic term for value: "
+                                + suggestedWorkSharingApproachAsFuzzyVariable.getLatestDefuzzifiedValue()
+                ));
+    }
 
-        if (linguisticTerm == null) {
-            throw new IllegalStateException("Unable to identify the linguistic term.");
+    /**
+     * Sets input parameters to the Fuzzy Inference System (FIS).
+     *
+     * @param fuzzyInferenceSystem           The FIS instance where input parameters will be set.
+     * @param slidingDecisionInputParameters The input parameters from the sliding decision request.
+     * @throws UnknownInputParameterException if an input parameter is not recognized by the FIS.
+     * @throws IllegalArgumentException       if the input parameter map is null or empty.
+     */
+    private void setInputParametersToFuzzyInterfaceSystem(FIS fuzzyInferenceSystem, Map<String, Object> slidingDecisionInputParameters) throws UnknownInputParameterException {
+        if (slidingDecisionInputParameters == null || slidingDecisionInputParameters.isEmpty()) {
```

Comment:
Should we maybe do this check already in the controller? This can be done totally independent of the concrete application scenario logic.